optuna/storages/_grpc/servicer.py

```diff
@@ -329,11 +347,16 @@ def GetTrials(
         study_id = request.study_id
         included_trial_ids = set(request.included_trial_ids)
         trial_id_greater_than = request.trial_id_greater_than
-        try:
-            trials = self._backend.get_all_trials(study_id, deepcopy=False)
-        except KeyError as e:
-            context.abort(code=grpc.StatusCode.NOT_FOUND, details=str(e))
-
+        if self._ttl_cache_seconds is None:
+            trials = self._get_all_trials(study_id, context)
+        else:
+            with self._lock:
+                now = time.time()
+                if now >= self._cached_trials[study_id].last_sync_time + self._ttl_cache_seconds:
+                    _logger.info("Syncing trials for study_id=%d", study_id)
```

Comment:
Nit: Users don't really need to be aware of the cache here, so I think logger.debug might be more appropriate. What do you think?

```suggestion
                    _logger.debug("Syncing trials for study_id=%d", study_id)
```